test/widgets/content_test.dart

```diff
@@ -442,18 +583,69 @@ void main() {
     testContentSmoke(ContentExample.emojiUnicodeLiteral);
   });
 
-  testContentSmoke(ContentExample.mathInline);
+  group('MathInlineNode', () {
+    testContentSmoke(ContentExample.mathInline);
+
+    testWidgets('maintains font-size ratio with surrounding text', (tester) async {
+      const html = '<span class="katex">'
+        '<span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>λ</mi></mrow>'
+          '<annotation encoding="application/x-tex"> \\lambda </annotation></semantics></math></span>'
+        '<span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.6944em;"></span><span class="mord mathnormal">λ</span></span></span></span>';
+      await checkFontSizeRatio(tester,
+        targetHtml: html,
+        targetFontSizeFinder: mkTargetFontSizeFinderFromPattern(r'\lambda'));
+    });
+  });
 
-  testWidgets('GlobalTime smoke', (tester) async {
-    // "<time:2024-01-30T17:33:00Z>"
-    await tester.pumpWidget(MaterialApp(home: BlockContentList(nodes: parseContent(
-      '<p><time datetime="2024-01-30T17:33:00Z">2024-01-30T17:33:00Z</time></p>'
-    ).nodes)));
+  group('GlobalTimeNode', () {
     // The time is shown in the user's timezone and the result will depend on
-    // the timezone of the environment running this test. Accept here a wide
+    // the timezone of the environment running these tests. Accept here a wide
     // range of times. See comments in "show dates" test in
     // `test/widgets/message_list_test.dart`.
-    tester.widget(find.textContaining(RegExp(r'^(Tue, Jan 30|Wed, Jan 31), 2024, \d+:\d\d [AP]M$')));
+    final renderedTextRegexp = RegExp(r'^(Tue, Jan 30|Wed, Jan 31), 2024, \d+:\d\d [AP]M$');
```

Comment:
This regexp only really makes sense when juxtaposed with the particular time value used in these tests. So if we're factoring out the regexp, let's factor out that time HTML too:
```suggestion
    final renderedTextRegexp = RegExp(r'^(Tue, Jan 30|Wed, Jan 31), 2024, \d+:\d\d [AP]M$');
    const timeSpanHtml = '<time datetime="2024-01-30T17:33:00Z">2024-01-30T17:33:00Z</time>';
```

(or perhaps better, HTML first and then comment and regexp)

This means the smoke test will splice that HTML inside a `<p>` element; but that's already pretty much what `checkFontSizeRatio` is going to do.